src/solvers/lowering/byte_operators.cpp

```diff
@@ -1802,6 +1806,7 @@ static exprt lower_byte_update_struct(
 
       byte_update_exprt bu = src;
       bu.set_op(lower_byte_extract(byte_extract_expr, ns));
+      bu.type() = bu.op().type();
```

Comment:
Perhaps this should always be a side-effect of `set_op`, much as the constructor copies its type?

Reply:
Fair point, I'll create a separate PR for that.